File: test_tools.js
const puppeteer = require('puppeteer');
const fs = require('fs');
const path = require('path');

async function testTools() {
    const toolsDir = path.join(__dirname, 'src', 'tools');
    const files = fs.readdirSync(toolsDir).filter(f => f.endsWith('.html') && f !== 'pdf-to-word.html' && f !== 'word-to-pdf.html');
    
    console.log(`Found ${files.length} tools to test.`);
    
    const browser = await puppeteer.launch({ headless: 'new' });
    const page = await browser.newPage();
    
    for (const file of files) {
        let hasError = false;
        page.on('console', msg => {
            if (msg.type() === 'error') {
                console.log(`[${file}] Console Error:`, msg.text());
                hasError = true;
            }
        });
        
        page.on('pageerror', err => {
            console.log(`[${file}] Page Error:`, err.toString());
            hasError = true;
        });

        try {
            await page.goto(`http://localhost:8080/tools/${file}`, { waitUntil: 'networkidle0', timeout: 10000 });
        } catch (e) {
            console.log(`[${file}] Navigation Error:`, e.message);
        }
        
        // Remove listeners for the next file
        page.removeAllListeners('console');
        page.removeAllListeners('pageerror');
    }
    
    await browser.close();
    console.log('Testing complete.');
}

testTools();

File: test_tools.py
import os
import sys
from playwright.sync_api import sync_playwright

def run():
    tools_dir = os.path.join('src', 'tools')
    if not os.path.exists(tools_dir):
        print(f"Directory {tools_dir} not found")
        sys.exit(1)
        
    files = [f for f in os.listdir(tools_dir) if f.endswith('.html') and f not in ('pdf-to-word.html', 'word-to-pdf.html')]
    print(f"Found {len(files)} tools to test.")
    
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        
        for f in files:
            errors = []
            page.on("console", lambda msg: errors.append(msg.text) if msg.type == "error" else None)
            page.on("pageerror", lambda exc: errors.append(str(exc)))
            
            try:
                page.goto(f"http://localhost:8080/tools/{f}", wait_until="networkidle", timeout=10000)
            except Exception as e:
                print(f"[{f}] Timeout or Navigation Error: {e}")
                
            if errors:
                for err in errors:
                    # Ignore harmless things
                    if "favicon" not in err.lower() and "analytics" not in err.lower() and "adsterra" not in err.lower() and "404" not in err:
                        print(f"[{f}] Error: {err}")
            
            page.close()
            page = browser.new_page()
            
        browser.close()
        print("Testing complete.")

if __name__ == '__main__':
    run()
